lib/rubocop/cop/style/instance_vars_in_class.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for instance variables used in classes outside of the
+      # initializer. It encourages using attr_reader or attr_accessor to set
+      # up safer methods to read from or write to your instance variable.
+      #
+      # In Rails projects, it's probably good to exclude controllers, jobs,
+      # and mailers straightaway.
+      #
+      # ```yaml
+      # Exclude:
+      #   - "app/controllers/**/*"
+      #   - "app/jobs/**/*"
+      #   - "app/mailers/**/*"
+      # ```
+      #
+      # @example
+      #
+      #   # bad
+      #   class Foo
+      #     def initialize
+      #       @foo = "bar"
+      #     end
+      #
+      #     def instance_method
+      #       @foo
+      #     end
+      #   end
+      #
+      #   # bad
+      #   class Foo
+      #     def initialize
+      #       @foo = "bar"
+      #     end
+      #
+      #     def instance_method
+      #       @foo = "baz"
+      #     end
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     attr_reader :foo
+      #
+      #     def initialize
+      #       @foo = "bar"
+      #     end
+      #
+      #     def instance_method
+      #       foo
+      #     end
+      #   end
+      #
+      #   # good
+      #   class Foo
+      #     attr_accessor :foo
+      #
+      #     def initialize
+      #       @foo = "bar"
+      #     end
+      #
+      #     def instance_method
+      #       foo = "baz"
```

Comment:
This doesn't work as `Foo#foo=`, but just as a local var assignment. This has to be `self.foo = "baz"` instead

Reply:
Thanks for catching!